torch/_inductor/codecache.py

```diff
@@ -402,7 +522,7 @@ def __bool__(self):
         if config.cpp.vec_isa_ok is not None:
             return config.cpp.vec_isa_ok
 
-        key, input_path = write(VecISA._avx_code, "cpp")
+        key, input_path = write(VecISA._avx_code, code_hash(VecISA._avx_code), "cpp")
```

Comment:
You are always calling write(src, code_hash(src), ...) wondering if it would be better to factor this into the write interface somehow

Reply:
improved the api